test/miscellaneous_tests/compound_macro.jl

```diff
@@ -0,0 +1,23 @@
+### Fetch Packages and Set Global Variables ###
+using Catalyst, ModelingToolkit, Test
+@variables t
+
+let 
+    @parameters k
+    @species H(t) O(t)
+    @compound H2O(t) 1H 2O
+    @test typeof(H2O) == Num
+    arr_components = [1H ,2O]
+    @test all((string(c1) == string(c2) for (c1, c2) = zip(components(H2O), arr_components)))
```

Comment:
You should test equality of the numeric coefficients and the symbolic variables separately. For symbolic variables you can use `isequal(A,B)` to test if they are the same symbolic. You shouldn't ever be converting to strings when working with symbolic variables except for display reasons.